src/routes/Profile/cache.ts

```diff
@@ -0,0 +1,10 @@
+import { GetProfileBadgesResponse } from '@graphql';
+
+let SEASON_BADGE_CACHE: Record<string, GetProfileBadgesResponse | undefined> = {};
```

Comment:
do you think it's worth considering caching this data to persist even after a reload? i think that can potentially greatly improve the UX.

Reply:
not really... too much caching can lead to bugs :D I thought about it but then we need to have caching per periods. Like 10min caching and I think people want data to be as live as possible. To go on traders do a trade come to profile and see their badge.

Reply:
yeah, that's the ideal experience we want to achieve, there are quite a few things that feed into this e.g. time it takes for the data to propagate through the subgraph, i think we need to strike a balance with caching (combined with deliberate append/delete logic that consistently keeps the cache in sync

Reply:
e.g. if we do a trade in the trading pool against an example position, we know for sure that we don't need to update the cache of any other position except of the position that was traded on (note there is an assumption baked into this thought, i.e. that all state changing transactions come from the UI, currently it's mostly true, though in the future we might see more transactions coming to voltz via alternative frontends/api integrators

Reply:
For me as enduser when I am on the page of Profile I don't want it to jump when I switch seasons all the time. But if I come from another page to the profile then it kinda makes sense to be refreshed from ground up.

What is your user experience that you are after?

Reply:
my expectation as a user is to not have a full ground up reload after returning to the page, minimises overall waiting time, though curious what others think of this UX

Reply:
might be worth discussing with Brokoli as well